### PR TITLE
Expose pretty-dom utility

### PR DESCRIPTION
**Pull request by gnapse**
**What**:

This exposes the already existing internal utility function to print out a DOM in a human-friendly way.

**Why**:

As it was discussed in #24, this can be useful for users to print out the DOM under a certain html element, to debug more easily when writing tests.

**How**:

Simply moved the already existing utility to its own module.

**Checklist**:

* [x] Documentation
* [x] Tests
* [x] Code coverage

Code coverage is not full now because of the `logDOM` utility function. Now in hindsight I'm not sure if it's a good idea to provide it. This is because there's no eslint rule now warning the users that they left this command in use in their tests, when I think the intent is that this is only to be used while creating and changing tests, but not meant to be left in actual committed code.

If you still think we should provide it, how do you think we should resolve the code coverage issue? Should we create a new test that uses it? Wouldn't that pollute the tests output? Is there a way to ignore some lines in the coverage report?

### CODE REVIEW

File: src/pretty-dom.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const {DOMElement, DOMCollection} = prettyFormat.plugins
+
+function prettyDOM(htmlElement) {
+  const debugContent = prettyFormat(htmlElement, {
+    plugins: [DOMElement, DOMCollection],
+    printFunctionName: false,
+    highlight: true,
+  })
+  const maxLength = process.env.DEBUG_PRINT_LIMIT || 7000
```
Comment (kentcdodds) [inline src/pretty-dom.js]: Perhaps we should accept this as an argument? I'm thinking that folks who are logging things themselves might want to control this more easily and in general probably want to see the whole thing. So perhaps we should only have this default for where it was already in use and the real default should be Infinity?

**Comment by kentcdodds** [comment]
As for the coverage thing:

```javascript
jest.spyOn(console, 'log').mockImplementation(() => {})
logDOM(document.createElement('div'))
expect(console.log.mock.calls).toMatchSnapshot()
console.log.mockRestore()
```

:smile:

**Comment by gnapse** [comment]
Yes, I was aware that documentation is pending but wanted to kick off the discussion of code here. I'll change the approach to the limit. I also wanted to discuss the code coverage issue plus the potential linting issue with `logDOM` that I mentioned.

**Comment by gnapse** [comment]
@kentcdodds what do you think about not providing `logDOM`? It's not about the coverage, but about the fact that it'd be effectively hiding a `console.log` from the user's eslint. A `console.log` that can potentially end up being committed into their vcs.

As a reference, I think I recall that enzyme's pretty print helper also did not do the `console.log` itself, it just gives you the output to for the user to log it out or whatever. It's also yet another API to document and maintain, when all it does it very simple.

File: src/pretty-dom.js (hunk 1 above)
Comment (antoaravinth) [reply src/pretty-dom.js]: Yes I agree with @kentcdodds. Currently when a test case is failing, its chipping of the container and I need to increase the `DEBUG_PRINT_LIMIT` to see the whole dom, which is causing me to re-run the case again.

Comment (gnapse) [reply src/pretty-dom.js]: I changed it, and now the env var is only used internally when pretty printing out DOM inside thrown exceptions by the element getters. The exposed `prettyDOM` does not limit the output at all, and allows a second optional argument for the user to set a limit per call.

**Comment by kentcdodds** [comment]
I'd be fine with not including `logDOM` :+1:

File: src/pretty-dom.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import prettyFormat from 'pretty-format'
+
+const {DOMElement, DOMCollection} = prettyFormat.plugins
+
+function prettyDOM(htmlElement, maxLength = undefined) {
```
Comment (kentcdodds) [inline src/pretty-dom.js]: Don't need the `= undefined` because that's functionally equivalent to what the value of `maxLength` would be anyway :)

Comment (gnapse) [reply src/pretty-dom.js]: I know, I just like to be more explicit in the declaration that the argument is not expected, to anyone that passes by and read the code. Guess flow or typescript would be clearer for that purpose but I have yet to learn to use one of them, it's on my list ;)

**Comment by jlongster** [comment]
Wow, that was fast! Thanks all!

**Pull request merged**